controllers/nodefeaturediscovery_controls.go

```diff
@@ -317,6 +317,13 @@ func DaemonSet(n NFD) (ResourceStatus, error) {
 		}
 		portFlag := fmt.Sprintf("--port=%d", port)
 		obj.Spec.Template.Spec.Containers[0].Args = []string{portFlag}
+
+		// check if running as instance
+		// https://kubernetes-sigs.github.io/node-feature-discovery/v0.8/advanced/master-commandline-reference.html#-instance
+		if n.ins.Spec.Instance != "" {
+			instanceFlag := fmt.Sprintf("--instance=%s", n.ins.Spec.Instance)
+			obj.Spec.Template.Spec.Containers[0].Args = []string{portFlag, instanceFlag}
```

Comment:
nit: I think it would be neater and more future-proof (if we add more flags) to use `append()` here